config/config.go

```diff
@@ -450,6 +451,13 @@ func (s *HeaderMatch) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return nil
 }
 
+func (h HTTPProbe) GetFailIfBodyTooLarge() bool {
```

Comment:
Alternatively I could do the `nil` check in `UnmarshalYAML`, but then it could panic if someone changes the variable under the hood later on.

Reply:
To preserve the current behaviour, `FailIfBodyTooLarge` should be set to `true` by default.

you should be able to set it to `true` by default in DefaultHTTPProbe at https://github.com/streamonkey/blackbox_exporter/blob/master/config/config.go#L51


if we make this bool instead of *bool, we do't have to worry about `nil`, and can remove this whole function.